src/Database/LSMTree/Internal/BlobRef.hs

```diff
@@ -71,7 +74,7 @@ data WeakBlobRef m h = WeakBlobRef {
 -- using 'releaseBlobRef' when no longer in use (e.g. after completing I\/O).
 --
 data StrongBlobRef m h = StrongBlobRef {
-      strongBlobRefFile :: {-# NOUNPACK #-} !(BlobFile m h)
+      strongBlobRefFile :: {-# NOUNPACK #-} !(Ref (BlobFile m h))
     , strongBlobRefSpan :: {-# UNPACK #-}   !BlobSpan
     }
   deriving stock (Show)
```

Comment:
Do we still need `StrongBlobRef` and `WeakBlobRef`? Could we now replace `WeakBlobRef m h` by `WeakRef (RawBlobRef m h)`, and similarly replace `StrongBlobRef m h` by `Ref (RawBlobRef m h)`?

Reply:
We can't use `WeakRef (RawBlobRef m h)` because `RawBlobRef m h` itself is not a reference counted type.

Unfortunately these things are not perfectly composable. This is because we can only have refs to types that expect to be reference counted (that contain a ref count and are in the right class).